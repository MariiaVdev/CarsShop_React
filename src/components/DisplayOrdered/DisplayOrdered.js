import styles from './DisplayOrdered.module.scss';
import PropTypes from 'prop-types';


const DisplayOrdered = (props) => {
    const { name, price, img, code, color } = props;

    return (

        <div className={styles.cardWrapper}>
            <img src={img} alt={name}></img>
            <div className={styles.cardName}>
                <span><b>{name}, </b></span>
                <span>Vendor code: {code}, </span>
                <span> Color: <button className={styles.cardBtnColor} style={{ backgroundColor: color }}></button>,  </span>
                <span><b>${price}</b></span>
            </div>            
        </div>

    )
}

DisplayOrdered.propTypes = {
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    code: PropTypes.number,
    color: PropTypes.string
}

DisplayOrdered.defaultProps = {
    code: 1234,
    color: 'grey'
}

export default DisplayOrdered;