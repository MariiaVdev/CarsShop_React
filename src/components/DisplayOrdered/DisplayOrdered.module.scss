.cardWrapper {
  display: grid;
  grid-template-columns: 80px 600px;
  box-sizing: border-box;
  padding: 5px;
  margin: 5px 5px;
  border: 1px solid lightgrey;
  border-radius: 5px;

  img {
    width: 75px;
    height: 50px;
  }

  .cardName {
    font-size: 16px;
    width: 400px;
    padding-top: 14px;

    span {
      font-size: 16px;
      padding-bottom: 5px;
    }
  }
  .cardBtnColor {
    width: 20px;
    height: 20px;
    border: 1px lightgrey solid;

  }

}