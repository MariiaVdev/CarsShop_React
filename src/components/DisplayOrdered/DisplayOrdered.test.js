import { render } from "@testing-library/react"
import DisplayOrdered from './DisplayOrdered'


describe("Render DisplayOrdered", () => {
    test("should DisplayOrdered render with default props", () => {
        const { asFragment } = render(<DisplayOrdered name={'test'} price={12345} img={'img'} />);
        expect(asFragment()).toMatchSnapshot()
    })
    test("should DisplayOrdered render with custom props", () => {
        const { asFragment } = render(<DisplayOrdered name={'test'} price={12345} img={'img'} code={1234} color={'red'} />);
        expect(asFragment()).toMatchSnapshot()
    })
})


