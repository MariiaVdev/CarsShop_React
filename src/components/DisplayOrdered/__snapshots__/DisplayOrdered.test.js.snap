// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DisplayOrdered should DisplayOrdered render with custom props 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <img
      alt="test"
      src="img"
    />
    <div
      class="cardName"
    >
      <span>
        <b>
          test, 
        </b>
      </span>
      <span>
        Vendor code: 1234, 
      </span>
      <span>
         Color: 
        <button
          class="cardBtnColor"
          style="background-color: red;"
        />
        ,  
      </span>
      <span>
        <b>
          $12345
        </b>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render DisplayOrdered should DisplayOrdered render with default props 1`] = `
<DocumentFragment>
  <div
    class="cardWrapper"
  >
    <img
      alt="test"
      src="img"
    />
    <div
      class="cardName"
    >
      <span>
        <b>
          test, 
        </b>
      </span>
      <span>
        Vendor code: 1234, 
      </span>
      <span>
         Color: 
        <button
          class="cardBtnColor"
          style="background-color: grey;"
        />
        ,  
      </span>
      <span>
        <b>
          $12345
        </b>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
